### DISCUSSION TITLE
simplify3d

### DISCUSSION THREAD

**Discussion by Senthiele** [jschuh/klipper-macros#122] [Q&A]
anyone have the starting info for simplify3d scripts and ending one also. new update is not working with old scripts

here is my old script

M190 S0
M109 S0
PRINT_START EXTRUDER=[extruder0_temperature] BED=[bed0_temperature] MESH_MIN=[build_min_x],[build_min_y] MESH_MAX=[build_max_x],[build_max_y]  LAYERS=[total_layer_count] NOZZLE_SIZE="0.4" FILAMENT="PLA"

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically purge (e.g. if
; using a Mosaic Palette, which requires the slicer to generate the purge).

**Comment by jschuh** [comment]
Are you asking how to split this out for the phased startup? If so, you just need to replace `PRINT_START` with `_PRINT_START_PHASE_INIT` and then add the extra lines. So, something this:

```
M190 S0
M109 S0
_PRINT_START_PHASE_INIT EXTRUDER=[extruder0_temperature] BED=[bed0_temperature] MESH_MIN=[build_min_x],[build_min_y] MESH_MAX=[build_max_x],[build_max_y] LAYERS=[total_layer_count] NOZZLE_SIZE="0.4" FILAMENT="PLA"
; Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
_PRINT_START_PHASE_PROBING
; Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE
```
